WalletWasabi.Fluent/ViewModels/Wallets/Send/TransactionPreviewViewModel.cs

```diff
@@ -440,6 +442,7 @@ private async Task OnConfirmAsync()
 				var finalTransaction =
 					await GetFinalTransactionAsync(transactionAuthorizationInfo.Transaction, _info);
 				await SendTransactionAsync(finalTransaction);
+				_wallet.UpdateUsedHdPubKeysLabels(transaction.HdPubKeysWithNewLabels);
```

Comment:
@Szpoti @molnard 

I still think this is not what we should do. Currently, whenever a transaction is sent, we have to manually call a method on the UI side to keep the labels in the correct form. 

My problem with this,
- What guarantees that the future developer knows this and won't forget to call that method?
- Updating business-related terms on the UI side is not correct.

Why doesn't it happen automatically? UI should only call 1 method, which is the SendTransaction method, everything else that is related to it should happen inside of that call.

As far as I see the main problem is the whole transaction building logic was for 1.0, where basically whenever you built a transaction that was sent too. In 2.0 this is not the case, we built transactions many many times that will never be sent. Maybe it would be worth reviewing the whole transaction building/broadcasting logic and updating it to the use case of 2.0.